binary_search_tree/tree.py

```diff
@@ -14,35 +14,122 @@ class Tree:
     def __init__(self):
         self.root = None
 
-    # Time Complexity: 
-    # Space Complexity: 
+    # Time Complexity: O(log(n))
+    # Space Complexity: O(n)
     def add(self, key, value = None):
-        pass
-
-    # Time Complexity: 
-    # Space Complexity: 
+        if self.root == None:
+            self.root = TreeNode(key, value)
+            return
+        
+        current = self.root
+        while current != None:
+            if current.key > key:
+                if current.left == None:
+                    current.left = TreeNode(key, value)
+                    return
+                else:
+                    current = current.left
+            else:
+                if current.right == None:
+                    current.right = TreeNode(key, value)
+                    return
+                else:
+                    current = current.right
+    
+    # Time Complexity: o(1)
+    # Space Complexity: O(n)
```

Comment:
since you aren't recursively calling this, the space complexity is O(1) because no new memory is dependent upon the size of the input